java/vector/src/main/java/org/apache/arrow/vector/complex/AbstractStructVector.java

```diff
@@ -233,6 +233,11 @@ public <T extends FieldVector> T getChild(String name, Class<T> clazz) {
 
   protected ValueVector add(String childName, FieldType fieldType) {
     FieldVector vector = fieldType.createNewSingleVector(childName, allocator, callBack);
+
+    if (conflictPolicy == ConflictPolicy.CONFLICT_REPLACE && vectors.containsKey(childName)) {
+      vectors.getAll(childName).forEach(c -> c.close());
+    }
```

Comment:
This address test failure in `TestStructVector`. When duplicate field names exist in `initializeChildrenFromFields` call, the initial offset buffer in previous vectors will cause memory leak. So these duplicate vectors are closed here.

Reply:
Shouldn't this be done in `putVector` instead?